src/main/java/org/passay/PropertiesMessageResolver.java

```diff
@@ -45,7 +45,15 @@ public PropertiesMessageResolver(final Properties properties)
   @Override
   protected String getMessage(final String key)
   {
-    return messageProperties.getProperty(key);
+    String msg = messageProperties.getProperty(key);
+    if (msg == null && key.contains(".")) {
+      String newKey = key;
+      while (msg == null && newKey.contains(".")) {
```

Comment:
Implementing this as a loop suggests that you intend to support multiple dots, which is not what we discussed. What's your thinking here?

Reply:
Seemed like a general improvement to support hierarchical messages. What's the downside?

Reply:
None, just wanted to poke at what you were thinking a bit.